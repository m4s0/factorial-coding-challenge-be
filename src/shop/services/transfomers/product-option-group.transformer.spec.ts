import { ProductOption } from '@Shop/entities/product-option.entity';
import { ProductOptionGroup } from '@Shop/entities/product-option-group.entity';
import { transformProductOptionGroup } from './product-option-group.transformer';

describe('transformOptionGroup', () => {
  it('should transform product option group correctly', () => {
    const productOptionGroup = {
      id: '123',
      name: 'Size',
      displayName: 'Select Size',

      options: [] as ProductOption[],
      createdAt: new Date('2024-01-01'),
      updatedAt: new Date('2024-01-01'),
    } as ProductOptionGroup;

    const result = transformProductOptionGroup(productOptionGroup);

    expect(result).toMatchSnapshot();
  });
});
