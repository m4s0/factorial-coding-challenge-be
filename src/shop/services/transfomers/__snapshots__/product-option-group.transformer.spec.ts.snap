// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformOptionGroup should transform product option group correctly 1`] = `
{
  "createdAt": "2024-01-01T00:00:00.000Z",
  "displayName": "Select Size",
  "id": "123",
  "name": "Size",
  "options": [],
  "updatedAt": "2024-01-01T00:00:00.000Z",
}
`;
